keanu-project/src/main/java/io/improbable/keanu/vertices/VertexId.java

```diff
@@ -17,13 +17,26 @@
 public class VertexId implements Comparable<VertexId> {
 
     public static final AtomicLong ID_GENERATOR = new AtomicLong(0L);
+    private static final int DEPTH_ONE_ARRAY_SIZE = 1;
 
-    long[] idValues = new long[1];
+    long[] idValues = new long[DEPTH_ONE_ARRAY_SIZE];
 
     public VertexId() {
         idValues[0] = ID_GENERATOR.getAndIncrement();
     }
 
+    public void increaseDepth(VertexId prefix) {
+        long[] newIdValues = new long[idValues.length + prefix.idValues.length];
+        System.arraycopy(prefix.idValues, 0, newIdValues, 0, prefix.idValues.length);
+        System.arraycopy(idValues, 0, newIdValues, prefix.idValues.length, idValues.length);
+        idValues = newIdValues;
+    }
+
+    public void resetID() {
+        idValues = new long[DEPTH_ONE_ARRAY_SIZE];
```

Comment:
Can a model be nested inside a nested model? Should the output vertices always get reset to depth 1 in that case?

Reply:
Yep - a model can be nested within another - but with how everything is set up at the moment you can only really operate on the "outermost" graph at any particular point (so when you call ModelComposition methods you're operating by necessity on the outermost graph).